en/04_Changelogs/beta/4.12.0-beta1.md

```diff
@@ -133,7 +133,67 @@ jQuery has made some breaking behavioural changes over time. If you are experien
+  onmatch: function() {
+    this.data('some-data', 'some-value');
+  }
+});
+$('.some-other-class').entwine({
+  // Add an instance of the above element into the DOM and then try to use the initialised data
+  onclick: function() {
+    this.parent(parentSelector).append('<div class="some-class"></div>');
+    // Expects the data to have been set in onmatch instantly because the event used to be synchronous
+    const initiliasedData = this.siblings('.some-class').data('some-data');
+    console.log(initiliasedData); // this will almost certainly print "undefined" to the console, because the MutationObserver event hasn't fired yet
+  }
+});
+```
+
+**After:**
+```js
+$('.some-class').entwine({
+  // Add an entwine property to defer initialisation events
+  DeferredInitilisationCallback: null,
+  // Add an entwine property to indicate whether this element has been initialised
+  IsInitialised: false,
+  // Initialise something in the onmatch or onadd handler
+  onmatch: function() {
+    this.data('some-data', 'some-value');
+    this.setIsInitialised(true);
+    // If some action was deferred, perform that action now
+    const deferredCallback = this.getDeferredInitilisationCallback();
+    if (typeof deferredCallback === 'function') {
+      deferredCallback();
+      this.setDeferredInitilisationCallback(null);
+    }
+  }
+});
+// Add an instance of the above element into the DOM and then try to use the initialised data
+$('.some-other-class').entwine({
+  onclick: function() {
+    this.parent(parentSelector).append('<div class="some-class"></div>');
+    const newElement = this.siblings('.some-class');
+    // Check if the element has been initialised before performing the action
+    if (newElement.getIsInitialised()) {
+      const initiliasedData = newElement.data('some-data');
+      console.log(initiliasedData); // this will almost never be reached but you might get a situation where the event fires immediately, so it pays to be safe
+    } else {
+      // Defer the action until the element has been initialised
+      newElement.setDeferredInitilisationCallback(function() {
+        const initiliasedData = newElement.data('some-data');
+        console.log(initiliasedData); // this will print "some-data" to the console, because it will be called after initialisation has finished
+      });
+    }
+  }
+});
+```
+
+Note that if you are not trying to used the initialised data/functionality _instantly_ after adding the element to the DOM, this kind of change is unnecessary (e.g. click event handlers which rely on _the element the user is clicking_ having been initialised do not need to defer their actions like this, because the user is extremely unlikely to click the element in the time it takes for the `MutationObserver` to fire its events). It is very much exclusive to scenarios where code which adds or replaces elements in the DOM then immediately expects some initialisation to have occurred.
```

Comment:
```suggestion
Note that if you are not trying to use the initialised data/functionality _immediately_ after adding the element to the DOM, this kind of change is not necessary (i.e. click event handlers which rely on _the element the user is clicking_ having been initialised do not need to defer their actions like this, because the user is extremely unlikely to click the element in the time it takes for the `MutationObserver` to fire its events). It is very much exclusive to scenarios where code which adds or replaces elements in the DOM then immediately expects some initialisation to have occurred.
```